### PR TITLE
chore: add definitions for findComponent and findAllComponents

### LINKED ISSUE #1529: New type definitions for TypeScript are missed
### Version
1.0.0

It looks `index.d.ts` isn't updated yet. After upgrading to `v1.0.0` I started to get deprecation errors, but methods like `findAllComponents` and `findComponent` are missed in `index.d.ts`.

### PR DESCRIPTION
**Pull request by lmiller1990**
**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Code style update
- [x] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] It's submitted to the `dev` branch.
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] All tests are passing: https://github.com/vuejs/vue-test-utils/blob/dev/.github/CONTRIBUTING.md#development-setup
- [ ] New/updated tests are included

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

Add types for `findComponent` and `findAllComponents`, updated docs.

resolves #1529

### CODE REVIEW

File: docs/api/wrapper/findAllComponents.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+## findAllComponents
+
+Returns a [`WrapperArray`](../wrapper-array/).
```
Comment (afontcu) [inline docs/api/wrapper/findAllComponents.md]:
```suggestion
Returns a [`WrapperArray`](../wrapper-array/) of all matching Vue components.
```

File: docs/api/wrapper/find.md [hunk 2]
```diff
@@ -20,34 +20,19 @@ import Bar from './Bar.vue'
...
-const fooRef = wrapper.find({ ref: 'foo' })
-expect(fooRef.is(Foo)).toBe(true)
+const byId = wrapper.find('#bar')
+expect(byId.element.id).toBe('bar')
 ```
 
 - **Note:**
 
-  - When chaining `find` calls together, only DOM selectors can be used
+  - You may chain `find` calls together:
```
Comment (afontcu) [inline docs/api/wrapper/find.md:31]: 🎉

**Comment by dobromir-hristov** [comment]
Looks good

**Review by afontcu — APPROVED** [review]
As long as the ES version bump in `tsconfig.json` is intended and expected, LGTM 👍

**Pull request merged**